### PR TITLE
🐛 Resource validation for creating draft.

### PR DESCRIPTION
**Pull request by sreehari-aot**
# Issue Tracking

JIRA: https://aottech.atlassian.net/browse/FWF-1674
Issue Type: BUG

# Changes

- Application <-> Drafts transition glitch fix
- Draft create only after validating the resource

### CODE REVIEW

**Review by abhilash-aot — APPROVED** [review]
isValidResource condition check is added here  to check for any invalid form Resource error before triggering the draft create api. This will prevent additional empty draft creations. Need to add the test case to check the newly added resource condition match as well. Also the formatting update is done at relevant places. Overall PR functionality looks good .

File: forms-flow-web/src/components/Form/Item/View.js [hunk 1]
```diff
@@ -196,6 +196,11 @@ const View = React.memo((props) => {
     }
   };
 
+  useEffect(() => {
+    if (form._id && !error) setIsValidResource(true);
+    return () => setIsValidResource(false);
+  }, [error, form._id]);
```
Comment (abhilash-aot) [inline forms-flow-web/src/components/Form/Item/View.js:202]: Chcek for types of possible errors here. Form validation errors can be ignored for draft

Comment (sreehari-aot) [reply forms-flow-web/src/components/Form/Item/View.js:202]: We only expect this to run for the first time, so the error that we are concerned here would be `Resource not found` error.

Comment (abhilash-aot) [reply forms-flow-web/src/components/Form/Item/View.js:202]: Yes . True. The submission and related validation errors will be at submission error api.

**Review by abilpraju-aot — APPROVED** [review]
LGTM

**Pull request merged**